kevm-pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -39,30 +40,92 @@ class Method:
         sort: KSort
         arg_names: tuple[str, ...]
         arg_types: tuple[str, ...]
+        contract: Contract
         contract_name: str
         payable: bool
         signature: str
+        ast: dict | None
 
-        def __init__(self, msig: str, id: int, abi: dict, contract_name: str, sort: KSort) -> None:
+        def __init__(self, msig: str, id: int, abi: dict, ast: dict | None, contract: Contract, sort: KSort) -> None:
             self.signature = msig
             self.name = abi['name']
             self.id = id
             self.arg_names = tuple([f'V{i}_{input["name"].replace("-", "_")}' for i, input in enumerate(abi['inputs'])])
             self.arg_types = tuple([input['type'] for input in abi['inputs']])
-            self.contract_name = contract_name
+            self.contract = contract
+            self.contract_name = contract.name
             self.sort = sort
             # TODO: Check that we're handling all state mutability cases
             self.payable = abi['stateMutability'] == 'payable'
+            self.ast = ast
 
         @property
         def klabel(self) -> KLabel:
             args_list = '_'.join(self.arg_types)
             return KLabel(f'method_{self.contract_name}_{self.name}_{args_list}')
 
+        @cached_property
+        def qualified_name(self) -> str:
+            return f'{self.contract_name}.{self.signature}'
+
         @property
         def selector_alias_rule(self) -> KRule:
             return KRule(KRewrite(KEVM.abi_selector(self.signature), intToken(self.id)))
 
+        @cached_property
+        def is_setup(self) -> bool:
+            return self.name == 'setUp'
+
+        def up_to_date(self, digest_file: Path) -> bool:
+            if not digest_file.exists():
+                return False
+            text = digest_file.read_text()
+            try:
+                digest_dict = json.loads(text)
+            except JSONDecodeError:
+                _LOGGER.warning(f'json text: {text}')
+                exit(1)
```

Comment:
```suggestion
            digest_dict = json.loads(digest_file.read_text())
```

No need to catch this exception.

Reply:
I applied this change.  I was trying to figure out why we were failing to parse the digest file, but this crash seems to have stopped happening. I wonder if the old digest format of just one hash was somehow being cached.